### PR TITLE
Update happy.py | mem check for singularity containers

### PR DESCRIPTION
**Pull request by dcmonroe**
Running happy in singularity on a Linux cluster caused crashes despite ample allocated memory/cores. 

I sandboxed the .simg and rebulit locally to confirm that this fix works.

Changes proposed in this pull request:

- check if rapidtide is running in a singularity container
- if so, run `tide_util.findavailablemem()`

### CODE REVIEW

**Comment by bbfrederick** [comment]
Interesting.  The check I was doing shouldn't have been specific to Docker - I was just checking an environment variable that was set in the Dockerfile).  I guess I don't understand how environment variables get translated when a Docker container is converted to an .simg file.

It looks to me like your code redoes the check using Singularity appropriate variables, then runs the same memory checking/setting code I did for Docker.  Would you be averse to just expanding the try statement?

**Comment by bbfrederick** [comment]
For reasons that are not entirely clear, your PR fails my automated tests in a very weird way.  I'm going to accept it as is so I can work through why.

**Comment by dcmonroe** [comment]
> Would you be averse to just expanding the try statement to make it more general?

Not at all. Parsimony preferred!

> For reasons that are not entirely clear, your PR fails my automated tests in a very weird way.

I noticed this and could not follow the error easily on my end. 

Thanks for publishing such a useful tool!

**Comment by bbfrederick** [comment]
Ok, very weird.  I cleaned up a conflict and reran the tests - and if failed again.  If I had to guess, I think that CircleCI uses singularity containers to run the test environments, and it may be getting upset with user code trying to monkey with them.  I'm going to put in a check to NOT run your code if this is running in CircleCI.

File: rapidtide/workflows/happy.py
Comment (bbfrederick) [inline rapidtide/workflows/happy.py:1]: Added a better check for running in CircleCI

**Review by bbfrederick — APPROVED** [review]

**Comment by bbfrederick** [comment]
Ok, this now seems to work on bare metal, in a Docker container, and on CircleCI.  Can you sync with the latest version of the PR and see if is still works properly in Singularity?

**Comment by dcmonroe** [comment]
I have synced the latest version and confirm that it works as a singularity build on my end! 

Thank you!

**Comment by bbfrederick** [comment]
Awesome!  Thank you for finding this and figuring out how to fix it!

**Comment by bbfrederick** [comment]
I've merged it and I'm going to rev the prerelease version so it will build a container.

**Pull request merged**